### DISCUSSION TITLE
Microsoft Stream on Sharepoint

### DISCUSSION THREAD

**Discussion by rabbyn** [aclap-dev/video-downloadhelper#281] [Help and Q&A]
### Your question

Hello, does anyone managed to download from [Microsoft Stream on Sharepoint](https://learn.microsoft.com/en-us/stream/streamnew/new-stream) (https / Oauth2 so I can't really share a link to reproduce) ? the DASH Streaming video is recognized by when trying to download it gives the error message below 

`Error: Exit code: 1

    at rs.receive (chrome-extension://jmkaglaafmhbcpleggkmaliipiilhldn/background/main.js:1:13157)
    at chrome-extension://jmkaglaafmhbcpleggkmaliipiilhldn/background/main.js:1:42625`

![image](https://github.com/aclap-dev/video-downloadhelper/assets/72968975/80351f5c-5ba4-454b-b95e-297daac4dd53)

### Addon & CoApp version and details

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: edge
lang: en-US
license: accepted
key (only 16 first characters): 2a80913566e6425d
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Edg/123.0.0.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\guillaume.berthier",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by Mattmjs** [comment]
Is it similar to these I found from the google group. What do you get on these links.

This one worked
https://groups.google.com/g/video-downloadhelper-q-and-a/c/IqFzDN5OjlQ

This sort of worked, all the video is there but no audio
https://groups.google.com/g/video-downloadhelper-q-and-a/c/45hnkfOYd9c

**Reply by rabbyn** [reply]
Thanks @Mattmjs but I thought that purchasing a license+support for this video dowloadhelper would save me from drowning in pages of forum post/replies or having to be an expert of ffmpeg. I already spent 1 hour reading the forum threads you gave me above,  jumping from 1 to an other and it seems that I would have to spend an other couple of hours just to start getting some hint on that matter. That's not my job/specialty I'm afraid.

PS: if anybody succeeded to download video from **Microsoft Stream on Sharepoint** with an other tool than video dowloadhelper I would appreciate. Thanks

**Reply by Mattmjs** [reply]
@rabbyn  I wanted to know if those links are comparable to your sharepoint stream since you can't share it.
Each link goes to a Microsoft page, did you try to see if they work or not ?
Is that too much for you to do ?

**Reply by rabbyn** [reply]
Hi @Mattmjs , each video from **Microsoft Stream for Sharepoint** hit a Microsoft Sharepoint URL. I don't mind to provide a sample  URL but you'll need a token (Oauth2 authentication) to access it (I can provide a token with an expiration of 1h in PM if you want) 
...if that can be meaningful for you here is the [Hit details.csv](https://github.com/aclap-dev/video-downloadhelper/files/14555019/Hit.details.csv). Thanks

**Reply by Mattmjs** [reply]
Ok I see an mpd url, but I don't know how to copy it. I'd like to try that link.
Nevermind I got it.

**Reply by rabbyn** [reply]
yes indeed there is a mpd url in the [Hit details.csv](https://github.com/aclap-dev/video-downloadhelper/files/14555019/Hit.details.csv) I provided but what do you mean by "I don't know how to copy it" ? from the csv file you mean ?  in case of there is a problem with my csv file [here is the URL]( https://northcentralus1-mediab.svc.ms/transform/videomanifest?provider=spo&inputFormat=mp4&cs=fFNQTw&correlationId=9de313a1-20cf-1000-0c33-5b3513ca4aba&docid=https%3A%2F%2Fmicrosoft.sharepoint.com%2Fteams%2FPowerBIEnterpriseVoice%2F_api%2Fv2.0%2Fdrives%2Fb%21zc9Lso0qgky6pkrNQXx9tunmcTVbT01LpFl-LaoKBg3gZsMpV_LhQIPAdRqrpuyT%2Fitems%2F01COM7XAXV42MRQESXKJFIKC5ED6UUVRBQ%3Ftempauth%3DeyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvbWljcm9zb2Z0LnNoYXJlcG9pbnQuY29tQDcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsImlzcyI6IjAwMDAwMDAzLTAwMDAtMGZmMS1jZTAwLTAwMDAwMDAwMDAwMCIsIm5iZiI6IjE3MTAxMzYwMzQiLCJleHAiOiIxNzEwMTU3NjM0IiwiZW5kcG9pbnR1cmwiOiJjZ1IwOXJhblR6NmNmQy9wWVNHbzVqSElBaXh2YnN5RTIvNWVoL1FFVEswPSIsImVuZHBvaW50dXJsTGVuZ3RoIjoiMTg2IiwiaXNsb29wYmFjayI6IlRydWUiLCJjaWQiOiJvUlBqbmM4Z0FCQU1NMXMxRThwS3VnPT0iLCJ2ZXIiOiJoYXNoZWRwcm9vZnRva2VuIiwic2l0ZWlkIjoiWWpJMFltTm1ZMlF0TW1FNFpDMDBZemd5TFdKaFlUWXROR0ZqWkRReE4yTTNaR0kyIiwic2lnbmluX3N0YXRlIjoiW1wia21zaVwiXSIsIm5hbWVpZCI6IjAjLmZ8bWVtYmVyc2hpcHxndWlsbGF1bWUuYmVydGhpZXIzX2FkZWNjb2dyb3VwLmNvbSNleHQjQG1pY3Jvc29mdC5vbm1pY3Jvc29mdC5jb20iLCJuaWkiOiJtaWNyb3NvZnQuc2hhcmVwb2ludCIsImlzdXNlciI6InRydWUiLCJjYWNoZWtleSI6IjBoLmZ8bWVtYmVyc2hpcHwxMDAzMjAwMjQzYWJjYzc4QGxpdmUuY29tIiwic2lkIjoiOWZhMmVmYzctZjdjNi00YmY5LWI3OWUtMzYxMDRiOWQwZTU5IiwiYWNycyI6ImMyLGMzIiwidHQiOiIwIiwiaXBhZGRyIjoiMTQ3LjE2MS4xODAuMTkyIn0.GBrx3pMsdqpc63j5SCTWSETSt-fEoRLcYtKjfJZI9Ug%26version%3DPublished&psi=5ddaf013-b0e4-493f-a912-3c6ac7c18c7d&enhanceAudio=true&cTag=%22c%3A%7B1899E6F5-5712-4A52-850B-A41FA94AC430%7D%2C2%22&action=Access&part=index&format=dash&altManifestMetadata=eyJBdWRpb0NvZGVjSW5mbyI6ImFhYyIsIkF1ZGlvRnJhbWVTaXplIjoxMDI0LCJBdWRpb1NhbXBsaW5nUmF0ZSI6NDQxMDAsIkF1ZGlvU3RhcnRUaW1lSW4xMDBOYW5vIjo3Mjk5MzAsIkF1ZGlvU3RyZWFtRHVyYXRpb24xMDBOYW5vIjoxNjE1MjY0NDAwLCJCaXRyYXRlIjoxMTI2MzkyLCJDb250YWluZXJTdGFydFRpbWVJbjEwME5hbm8iOjAsIkNvbnRhaW5zQXVkaW8iOnRydWUsIkNvbnRhaW5zVmlkZW8iOnRydWUsIkNvbnRlbnRBd2FyZUJpdHJhdGUiOm51bGwsIkNvbnRlbnRDYXRlZ29yeSI6Im5vbl90ZWFtc19tZWV0aW5nIiwiQ3JlYXRpb25Vbml4VGltZXN0YW1wU2Vjb25kcyI6MTcwOTkzOTg4OCwiQ3J5cHRvZ3JhcGhpY0RhdGFDcmVhdGVkRGF0ZVVuaXgiOjE3MDk5Mzk4ODgsIkNyeXB0b2dyYXBoaWNEYXRhSVYiOiIweEM4RTQ0ODZDMEY5NzU3QTVCN0RFMEEyNDM5MDU4OTFCIiwiQ3J5cHRvZ3JhcGhpY0RhdGFLaWQiOiJmYjQxNmY0ZDI4OS40NDYyMjAuNDI2NjIiLCJEdXJhdGlvbjEwME5hbm8iOjE2MTYwMDAwMDAsIkZyYW1lU2l6ZSI6eyJoZWlnaHQiOjEwODAsIndpZHRoIjoxOTIwfSwiR29wU2l6ZSI6LTEsIkgyNjRMZXZlbCI6NDEsIkhhc0JGcmFtZXMiOmZhbHNlLCJIZWFkZXJPZmZzZXRJbkJ5dGVzIjoyNjYyOTY2MiwiSGVhZGVyU2l6ZSI6MTc2NDkxLCJNZXRhZGF0YVNvdXJjZUluZm9ybWF0aW9uIjp7IkF1ZGlvU3RyZWFtRHVyYXRpb25Tb3VyY2UiOjEsIkF1ZGlvU3RyZWFtUmVhZEJ1ZmZlclNpemVJbkJ5dGVzIjoxNjM4NCwiQ29udGFpbmVyRHVyYXRpb25Tb3VyY2UiOjEsIlZpZGVvQml0cmF0ZVNvdXJjZSI6MSwiVmlkZW9TdHJlYW1EdXJhdGlvblNvdXJjZSI6MSwiVmlkZW9TdHJlYW1SZWFkQnVmZmVyU2l6ZUluQnl0ZXMiOjE2Mzg0fSwiUGFja2VkRnJhbWVSYXRlIjp7IkhpZ2giOjMwLCJMb3ciOjF9LCJQcmVtaXVtVmlkZW9QcmV0cmFuc2NvZGVEYXRhIjp7IlByZW1pdW1Qcm9jZXNzaW5nU3RhdHVzIjoxfSwiUmVhbEJhc2VGcmFtZVJhdGUiOnsiSGlnaCI6MzAsIkxvdyI6MX0sIlVwZGF0ZWFibGVBdWRpb1RyYWNrcyI6W10sIlZlcnNpb24iOnsiX0J1aWxkIjotMSwiX01ham9yIjo2LCJfTWlub3IiOjAsIl9SZXZpc2lvbiI6LTF9LCJWaWRlb0NvZGVjSW5mbyI6ImgyNjQiLCJWaWRlb1ByZXRyYW5zY29kZURhdGEiOnsiQXVkaW9RdWFsaXR5SWRzIjoiYXVkbWVkLGF1ZGhpZ2giLCJTZWdtZW50Q291bnQiOjIsIlZpZGVvUXVhbGl0eUlkcyI6InYyNDBwLHY0ODBwLHY3MjBwLHYxMDgwcCJ9LCJWaWRlb1N0YXJ0VGltZUluMTAwTmFubyI6MCwiVmlkZW9TdHJlYW1EdXJhdGlvbjEwME5hbm8iOjE2MTYwMDAwMDB9&altTranscode=1&useScf=true&pretranscode=0&transcodeahead=0&enableCdn=1&hybridPlayback=false&oif=4kp&pn=hostApp-OnePlayer-Web) mentioned in the "hit details"

**Reply by Mattmjs** [reply]
Ok, I tried your mpd link, it had an invalid argument error in ffprobe. In another tool **yt-dlp** this is what I got below.

ERROR: [generic] videomanifest?provider=spo&inputFormat=mp4&cs=fFNQTw&correlationId=9de313a1-20cf-1000-0c33-5b3513ca4aba&docid=https://microsoft.sharepoint.com/teams/PowerBIEnterpriseVoice/_api/v2.0/drives/b!zc9Lso0qgky6pkrNQXx9tunmcTVbT01LpFl-LaoKBg3gZsMpV_LhQIPAdRqrpuyT/items/01COM7XAXV42MRQESXKJFIKC5ED6UUVRBQ?tempauth=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvbWljcm9zb2Z0LnNoYXJlcG9pbnQuY29tQDcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsImlzcyI6IjAwMDAwMDAzLTAwMDAtMGZmMS1jZTAwLTAwMDAwMDAwMDAwMCIsIm5iZiI6IjE3MTAxMzYwMzQiLCJleHAiOiIxNzEwMTU3NjM0IiwiZW5kcG9pbnR1cmwiOiJjZ1IwOXJhblR6NmNmQy9wWVNHbzVqSElBaXh2YnN5RTIvNWVoL1FFVEswPSIsImVuZHBvaW50dXJsTGVuZ3RoIjoiMTg2IiwiaXNsb29wYmFjayI6IlRydWUiLCJjaWQiOiJvUlBqbmM4Z0FCQU1NMXMxRThwS3VnPT0iLCJ2ZXIiOiJoYXNoZWRwcm9vZnRva2VuIiwic2l0ZWlkIjoiWWpJMFltTm1ZMlF0TW1FNFpDMDBZemd5TFdKaFlUWXROR0ZqWkRReE4yTTNaR0kyIiwic2lnbmluX3N0YXRlIjoiW1wia21zaVwiXSIsIm5hbWVpZCI6IjAjLmZ8bWVtYmVyc2hpcHxndWlsbGF1bWUuYmVydGhpZXIzX2FkZWNjb2dyb3VwLmNvbSNleHQjQG1pY3Jvc29mdC5vbm1pY3Jvc29mdC5jb20iLCJuaWkiOiJtaWNyb3NvZnQuc2hhcmVwb2ludCIsImlzdXNlciI6InRydWUiLCJjYWNoZWtleSI6IjBoLmZ8bWVtYmVyc2hpcHwxMDAzMjAwMjQzYWJjYzc4QGxpdmUuY29tIiwic2lkIjoiOWZhMmVmYzctZjdjNi00YmY5LWI3OWUtMzYxMDRiOWQwZTU5IiwiYWNycyI6ImMyLGMzIiwidHQiOiIwIiwiaXBhZGRyIjoiMTQ3LjE2MS4xODAuMTkyIn0: This video is DRM protected

**Reply by rabbyn** [reply]
ok so do you mean that video-downloadhelper can not do the job because "**This video is DRM protected**" ? any surprise here ? how are the other users doing to download protected content from other platforms? (If i'm not wrong everything is "protected" ...YouTube...ect)

**Reply by Mattmjs** [reply]
Yes correct, this means your content is protected by DRM unfortunately. This is not just an issue for VDH but all download tools &
software. Others aren't downloading protected content on other platforms since downloading is not possible for DRM protected content. 

Your only action now is to find screen recording software.
Also you can confirm for yourself it's protected using Firefox and playing the content, do you see the icon as in the image below:

![DRM icon](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/00bee824-44ca-4f4d-a97a-b5b7a31765c4)

**Reply by Mattmjs** [reply]
I'm a user like you , I don't work for the software in case you were wondering. Sorry you can't get the content.
Well you can if screen recording it.

**Reply by rabbyn** [reply]
Ok thanks for trying @Mattmjs . PS : I didn't know Youtube video's were not DRM protected. Anyway I'm going to request a refund then because this tool is useless for me.

**Reply by Mattmjs** [reply]
YouTube has a movies category, so that's why DRM is present. Normal user generated content isn't protected though.
